tests/unit/maxmemory.tcl

```diff
@@ -1,3 +1,140 @@
+start_server {tags {"maxmemory" "external:skip"}} {
+    r config set maxmemory 10mb
+    r config set maxmemory-policy allkeys-lru
+    set server_pid [s process_id]
+
+    proc init_test {client_eviction} {
+        r flushdb
+
+        set prev_maxmemory_clients [r config get maxmemory-clients]
+        if $client_eviction {
+            r config set maxmemory-clients 3mb
+        } else {
+            r config set maxmemory-clients 0
+        }
+
+        r config resetstat
+        # fill 5mb using 50 keys of 100kb
+        for {set j 0} {$j < 50} {incr j} {
+            r setrange $j 100000 x
+        }
+        assert_equal [r dbsize] 50
+    }
+    
+    proc verify_test {client_eviction} {
+        set evicted_keys [s evicted_keys]
+        set evicted_clients [s evicted_clients]
+        set dbsize [r dbsize]
+        
+        if $::verbose {
+            puts "evicted keys: $evicted_keys"
+            puts "evicted clients: $evicted_clients"
+            puts "dbsize: $dbsize"
+        }
+
+        if $client_eviction {
+            assert_morethan $evicted_clients 0
+            assert_equal $evicted_keys 0
+            assert_equal $dbsize 50
+        } else {
+            assert_equal $evicted_clients 0
+            assert_morethan $evicted_keys 0
+            assert_lessthan $dbsize 50
+        }
+    }
+
+    foreach {client_eviction} {false true} {
+        set clients {}
+        test "eviction due to output buffers of many MGET clients, client eviction: $client_eviction" {
+            init_test $client_eviction
+
+            for {set j 0} {$j < 20} {incr j} {
+                set rr [redis_deferring_client]
+                lappend clients $rr
+            }
```

Comment:
I run test on MacOS, keys evict will be triggered after execution here, causing the test to fail when `client_eviction` is `true`.
On macOS the max memory will be 10574672 bytes, while in ubuntu it is 9676648 bytes.

MacOS:
```
mem_used: 10574672
evict: 1
stat memory: 3080256
mem_used: 10541872
evict: 2
15569:M 01 Sep 2021 18:23:24.331 * Evicting client: id=6 addr=127.0.0.1:55904 laddr=127.0.0.1:21111 fd=14 name= age=0 idle=0 flags=N db=9 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=16890 argv-mem=0 multi-mem=0 obl=16752 oll=5 omem=491640 tot-mem=525944 events=r cmd=mget user=default redir=-1
```

Reply:
This might be an edge case. Can you run the same tests but with the following patch applied?
```diff
diff --git a/tests/unit/maxmemory.tcl b/tests/unit/maxmemory.tcl
index 7d55998fb..0bc0e4b6d 100644
--- a/tests/unit/maxmemory.tcl
+++ b/tests/unit/maxmemory.tcl
@@ -1,5 +1,5 @@
 start_server {tags {"maxmemory" "external:skip"}} {
-    r config set maxmemory 10mb
+    r config set maxmemory 11mb
     r config set maxmemory-policy allkeys-lru
     set server_pid [s process_id]
```

Reply:
Tested fine after changing to 11m.

Reply:
thanks. fixing.